scipy/signal/spectral.py

```diff
@@ -511,11 +511,406 @@ def spectrogram(x, fs=1.0, window=('tukey',.25), nperseg=256, noverlap=None,
+    See Also
+    --------
+    istft: Inverse Short Time Fourier Transform
+    check_COLA: Check whether the Constant OverLap Add (COLA) constraint is met
+    welch: Power spectral density by Welch's method.
+    spectrogram: Spectrogram by Welch's method.
+    csd: Cross spectral density by Welch's method.
+    lombscargle: Lomb-Scargle periodogram for unevenly sampled data
+
+    Notes
+    -----
+    In order to enable inversion of an STFT via the inverse STFT in
+    `istft`, the signal windowing must obey the constraint of "Constant
+    OverLap Add" (COLA), and the input signal must have complete windowing
+    coverage (i.e. ``(x.shape[axis] - nperseg) % (nperseg-noverlap) == 0``).
+
+    The COLA constraint ensures that every point in the input data is
+    equally weighted, thereby avoiding aliasing and allowing full
+    reconstruction. Whether a choice of `window`, `nperseg`, and `noverlap`
+    satisfy this constraint can be tested with `check_COLA`.
+
+    ..  versionadded:: 0.18.0
+
+    References
+    ----------
+    .. [1] Oppenheim, Alan V., Ronald W. Schafer, John R. Buck
+           "Discrete-Time Signal Processing", Prentice Hall, 1999.
+    .. [2] Daniel W. Griffin, Jae S. Limdt "Signal Estimation from Modified
+           Short Fourier Transform", IEEE 1984, 10.1109/TASSP.1984.1164317
+
+    Examples
+    --------
+    >>> from scipy import signal
+    >>> from scipy.fftpack import fftshift
+    >>> import matplotlib.pyplot as plt
+
+    Generate a test signal, a 2 Vrms sine wave whose frequency is slowly
+    modulated around 3kHz, corrupted by white noise of exponentially decreasing
+    magnitude sampled at 10 kHz.
+
+    >>> fs = 10e3
+    >>> N = 1e5
+    >>> amp = 2 * np.sqrt(2)
+    >>> noise_power = 0.01 * fs / 2
+    >>> time = np.arange(N) / float(fs)
+    >>> mod = 500*np.cos(2*np.pi*0.25*time)
+    >>> carrier = amp * np.sin(2*np.pi*3e3*time + mod)
+    >>> noise = np.random.normal(scale=np.sqrt(noise_power), size=time.shape)
+    >>> noise *= np.exp(-time/5)
+    >>> x = carrier + noise
+
+    Compute and plot the STFT's magnitude.
+
+    >>> f, t, Zxx = signal.stft(x, fs, nperseg=1000)
+    >>> f = fftshift(f)
+    >>> Zxx = fftshift(Zxx, axes=0)
+    >>> plt.pcolormesh(t, f, np.abs(Zxx), vmin=0, vmax=amp)
+    >>> plt.ylim([0, f[-1]])
+    >>> plt.title('STFT Magnitude')
+    >>> plt.ylabel('Frequency [Hz]')
+    >>> plt.xlabel('Time [sec]')
+    >>> plt.show()
+    """
+
+    freqs, time, Zxx = _spectral_helper(x, x, fs, window, nperseg, noverlap,
+                                        nfft=None, detrend=detrend,
+                                        return_onesided=False,
+                                        scaling='spectrum', axis=axis,
+                                        mode='stft', padded=padded,
+                                        centered=centered, strict=strict)
+
+    return freqs, time, Zxx
+
+
+def istft(Zxx, fs=1.0, window='hann', noverlap=128, centered=True, strict=True,
```

Comment:
The documentation specifies that `noverlap` is `None`per default, while it is set to `128` here.